tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -18,136 +17,146 @@
-                return;
-            }
+                Log.Information("Live Debugger initialization started");
+
+                _remoteConfigurationManager.RegisterProduct(Product);
 
-            if (_settings.TransportType != TracesTransportType.Default)
+                Product.ConfigChanged += (sender, args) => AcceptConfiguration(args);
+                AppDomain.CurrentDomain.AssemblyLoad += (sender, args) => CheckUnboundProbes();
+                AppDomain.CurrentDomain.DomainUnload += (sender, args) => _lineProbeResolver.OnDomainUnloaded();
+
+                await StartAsync().ConfigureAwait(false);
+            }
+            catch (Exception e)
             {
-                Log.Information("Live Debugger is not supported on UDS or named pipelines.");
-                return;
+                Log.Error(e, "Initializing Live Debugger failed.");
             }
 
-            Log.Information("Live Debugger initialization started");
-
-            Task.Run(() => InitializeAsync());
+            bool CanInitialize()
+            {
+                if (_isInitialized)
+                {
+                    return false;
+                }
 
-            AppDomain.CurrentDomain.AssemblyLoad += (sender, args) => CheckUnboundProbes();
-            AppDomain.CurrentDomain.DomainUnload += (sender, args) => _lineProbeResolver.OnDomainUnloaded();
+                if (!_settings.Enabled)
+                {
+                    Log.Information("Live Debugger is disabled. To enable it, please set DD_DEBUGGER_ENABLED environment variable to 'true'.");
+                    return false;
+                }
 
-            Log.Information("Live Debugger initialization completed");
-        }
+                if (_settings.TransportType != TracesTransportType.Default)
+                {
+                    Log.Information("Live Debugger is not currently supported when using Unix Domain Sockets or Named Pipes.");
+                    return false;
+                }
 
-        private async Task InitializeAsync()
-        {
-            try
-            {
-                var isDiscoverySuccessful = await _discoveryService.DiscoverAsync().ConfigureAwait(false);
-                var isProbeConfigurationSupported = isDiscoverySuccessful && !string.IsNullOrWhiteSpace(_discoveryService.ProbeConfigurationEndpoint);
-                if (_settings.ProbeMode == ProbeMode.Agent && !isProbeConfigurationSupported)
+                var isConfigurationSupported = !string.IsNullOrWhiteSpace(_discoveryService.ConfigurationEndpoint);
+                if (!isConfigurationSupported)
                 {
-                    Log.Warning("You must upgrade datadog-agent in order to leverage the Live Debugger. All debugging features will be disabled.");
-                    return;
+                    Log.Warning("Please ensure that your agent is support Remote Configuration management and it enabled.");
```

Comment:
```suggestion
                    Log.Warning("Live Debugger could not be enabled because Remote Configuration Management is not available. Please ensure that you are using datadog-agent version 7.38.0 or higher, and that Remote Configuration Management is enabled in datadog-agent's yaml configuration file.");
```